kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala

```diff
@@ -172,6 +172,13 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
     super.stop()
   }
 
+  def getUserName(hs2ProxyUser: String): String = {
+    val sessionConf = Option(hs2ProxyUser).filter(_.nonEmpty).map(proxyUser =>
+      Map(KyuubiAuthenticationFactory.HS2_PROXY_USER -> proxyUser)).getOrElse(Map())
+
+    getUserName(sessionConf)
+  }
+
   def getUserName(sessionConf: Map[String, String]): String = {
```

Comment:
we can catch the exception in this method.  `def getUserName(sessionConf: Map[String, String])`.

try {
...
} catch {
case t: Throwable =>
        throw new NotAllowedException(t.getMessage)
}